spi/control-plane/contract-spi/src/main/java/org/eclipse/edc/connector/controlplane/contract/spi/types/offer/ContractOffer.java

```diff
@@ -103,8 +104,17 @@ public Builder policy(Policy policy) {
 
         public ContractOffer build() {
             Objects.requireNonNull(contractOffer.id, "Id must not be null");
+            var offerId = ContractOfferId.parseId(contractOffer.id)
+                    .orElseThrow(failure -> new IllegalArgumentException(failure.getFailureDetail()));
             Objects.requireNonNull(contractOffer.assetId, "Asset id must not be null");
             Objects.requireNonNull(contractOffer.policy, "Policy must not be null");
+            if (!offerId.assetIdPart().equals(contractOffer.assetId)) {
+                throw new IllegalArgumentException("Asset id %s must be equal to asset id part of offer id %s".formatted(contractOffer.assetId, offerId.assetIdPart()));
+            }
```

Comment:
this check will cause issues in the communication with connectors that are not EDC (the contract offer id structure is not standardized, it is EDC specific)